game-core/src/main/java/games/strategy/triplea/ai/pro/ProPurchaseAi.java

```diff
@@ -2269,37 +2269,34 @@ private void upgradeUnitsWithRemainingPUs(
           break;
         }
 
-        // Determine best long range attack option (prefer air units)
-        ProPurchaseOption bestAttackOption = null;
-        double maxAttackEfficiency =
-            minPurchaseOption.getAttackEfficiency()
-                * minPurchaseOption.getMovement()
-                * minPurchaseOption.getCost()
-                / minPurchaseOption.getQuantity();
+        // Determine best upgrade option (prefer air units)
+        // TODO: ensure map has carriers or air unit has range to reach enemy land mass
```

Comment:
nit, missing the TODO "tag"

Reply:
Not really relevant here as this comment isn't directly related to this PR. Just something I noticed.

Reply:
In this case, a user name is maybe a good tag. I don't think anyone else can complete this TODO. 
More options:
- create an AI improvements project, can use the project number.
- Generic "AI-IMPROVEMENTS"  tag

Reply:
Given all the AI code is packaged together, I don't a TODO tag really adds anything as you can just search for all the TODOs in the AI code.

Reply:
Maybe this item should go to your personal backlog? Are you planning on fixing this imminently? Maybe "AI" is a good enough tag.

How can a person easily refine a search to just AI code? Should we be forced to go through such an exercise if looking for TODO statements that are 'orphaned'? At some point soon, all TODOs without a tag would/should probably get dropped. We've many like this where only the person that wrote them knows what needs to be done, sometimes even the original author doesn't even have that context anymore.

Reply:
Go to games.strategy.triplea.ai package. I'd like to think I could speak to all the TODOs in the AI code since I probably wrote all of them :)  Honestly for now, I'd say unless there is a PR/issue to tie a TODO to, I don't think adding some filler tag is worth it. At some point, if we do that clean up, I'm going to have to review all the existing AI TODOs anyways.

Reply:
> Go to games.strategy.triplea.ai package

If we do a search for TODOs at a top level, we'd still have to filter out the package. 

I'm also concerned we're eliding a convention. It seems "except for AI" code to be a big exception. IMO it would be pretty simple to add an AI tag, my concern would be very satisfied, and we'd easily know the new vs old TODOs within that code. Any that are still current would presumably get such a tag. I'm also concerned we could easily wind up with "yup, that's current, yup, that too. Will I ever get to any of them? Probably not. Can anyone else pick up this TODO and fix it? no". Seems like we would not be actually fixing the problem of dangling TODOs that we had decided we wanted to stop making worse.

Reply:
We can revisit the convention in this case. I've also found there are gaps to it. 
The convention as written was not meant to be an advisory, if so, it would have been a trivial suggestion, which is not something worth even writing down as a convention.

https://github.com/triplea-game/triplea/wiki/Code-Conventions#todo-comments-attach-a-tracking-token-5249


Here is an issue to revisit the convention: https://github.com/triplea-game/triplea/issues/5549

If we go forward, we could be saying that it's okay to push forward on conventions where we do not agree and then we revisit them. IMO it would be worth just placating a silly convention just for the sake that we can say that we follow conventions when they are written down.

With that said, our focus probably should be on issue #5549 to see how we can revisit the original problem we were trying to solve.